directories = ["testdiamond"]
sampleids = ["SRR9641933"]
lanes = [1,2]

project_ids = ["PRJNA548383", "PRJNA544061", "PRJNA508385"]
acc_ids = []

for project in project_ids:
    with open("../Accession_lists/"+project+"_Acc_List.txt", "r") as f:
        acc_ids += [line.strip() for line in f.readlines()]

"""
rule all: #for testing
    input: 
        expand("../testdiamond/{sampleid}/{sampleid}.assembled.fastq", sampleid=sampleids),
        expand("../testdiamond/{sampleid}/{sampleid}.discarded.fastq", sampleid=sampleids),
        expand("../testdiamond/{sampleid}/{sampleid}.unassembled.forward.fastq", sampleid=sampleids),
        expand("../testdiamond/{sampleid}/{sampleid}.unassembled.reverse.fastq", sampleid=sampleids),

        expand("../testdiamond/{sampleid}/diamond_results_{sampleid}.txt", sampleid=sampleids)

"""
rule all: #for fiji
    input:
        # merge_reads creates these
        expand("../data/{sampleid}/{sampleid}.assembled.fastq",
        sampleid=acc_ids),
        expand("../data/{sampleid}/{sampleid}.discarded.fastq",
        sampleid=acc_ids),
        expand("../data/{sampleid}/{sampleid}.unassembled.forward.fastq",
        sampleid=acc_ids),
        expand("../data/{sampleid}/{sampleid}.unassembled.reverse.fastq",
        sampleid=acc_ids),

        # trim_short_reads creates these
        expand("../data/{sampleid}/{sampleid}.assembled.75.fastq",
        sampleid=acc_ids)

        # diamond creates a results file
        expand("../data/{sampleid}/diamond_results_{sampleid}.txt",
        sampleid=acc_ids)

        # uniprot mapping creates this
        expand("../data/{sampleid}/diamond_results_{sampleid}_mapped.txt",
        sampleid=acc_ids)


"""
rule get_data:
    output:
        "../data/{sampleid}/{sampleid}_1.fastq",
        "../data/{sampleid}/{sampleid}_2.fastq",
        "../data/{sampleid}/{sampleid}/{sampleid}.sra",
    run:
        commands = [
        "prefetch {wildcards.sampleid}", 
        "fasterq-dump {wildcards.sampleid} --outdir ../data/{wildcards.sampleid} --progress"
        ]

        for c in commands:
            shell(c)
"""
# merge reads using PEAR
rule merge_reads:
    threads: 20
    input:
        "../data/{sampleid}/{sampleid}_1.fastq",
        "../data/{sampleid}/{sampleid}_2.fastq"
    output:
        "../data/{sampleid}/{sampleid}.assembled.fastq",
        "../data/{sampleid}/{sampleid}.discarded.fastq",
        "../data/{sampleid}/{sampleid}.unassembled.forward.fastq",
        "../data/{sampleid}/{sampleid}.unassembled.reverse.fastq"
    shell:
        """ 
        pear -f ../data/{wildcards.sampleid}/{wildcards.sampleid}_1.fastq \
        -r ../data/{wildcards.sampleid}/{wildcards.sampleid}_2.fastq \
        -o ../data/{wildcards.sampleid}/{wildcards.sampleid} \
        --threads 20
        """

# trim merged reads using custom script
rule trim_short_reads:
    input:
        "../data/{sampleid}/{sampleid}.assembled.fastq"
    output:
        "../data/{sampleid}/{sampleid}.assembled.75.fastq"
    shell: # structure is [input file] [ouput file] [minimum read length]
    """
    bash trim_short_reads.sh \
    ../data/{wildcards.sampleid}/{wildcards.sampleid}.assembled.fastq \
    ../data/{wildcards.sampleid}/{wildcards.sampleid}.assembled.75.fastq \
    75
    """

# run diamond on trimmed merged reads
rule run_diamond:
    threads: 20
    input:
        "../data/{sampleid}/{sampleid}.assembled.75.fastq"
    output:
        "../data/{sampleid}/diamond_results_{sampleid}.txt"
    shell:
        """
        diamond blastx --db ../database-building/uniprot-nicotine-db-fiji.dmnd \
        -q ../data/{wildcards.sampleid}/{wildcards.sampleid}.assembled.fastq \
        --threads 20 > ../data/{wildcards.sampleid}/diamond_results_{wildcards.sampleid}.txt
        """ 

# map diamond hits to uniprot gene name
rule map_uniprot:
    input:
        "../data/{sampleid}/diamond_results_{sampleid}.txt"
    output:
        "../data/{sampleid}/diamond_results_{sampleid}_mapped.txt"
    shell:
        """
        python map_uniprot_hits.py \
        -i ../data/{wildcards.sampleid}/diamond_results_{wildcards.sampleid}.txt \
        -o ../data/{wildcards.sampleid}/diamond_results_{wildcards.sampleid}_mapped.txt"
        """
